### ISSUE TITLE
Common DocFx [!NOTE] notation being flagged with verify links

### ISSUE THREAD

**Issue by MicroTuld** [microsoft/vscode-markdown-languageservice#164]
With DocFx, a common syntax is:
``` markdown
> [!NOTE]
> This is a special note box.
```
but verification is flagging this with "No link definition found" (link.no-such-reference). This seems to be because `()` is expected.

I would like to continue using link verification, but the only override available seems to be for what is inside the `()`, not the `[]`.

This issue seems to be similar to #152

**Comment by lucaspar** [comment]
Agreed - I suggest updating the issue title to cover all Markdown alerts, something like

> Markdown alerts cause false positives from link detection feature.

+ Other false-positive link detection issues:
    + #155 
    + https://github.com/microsoft/vscode/issues/150672
+ Markdown alerts:
    + [Discussion](https://github.com/orgs/community/discussions/16925)
    + [Description](https://learn.microsoft.com/en-us/contribute/content/markdown-reference#alerts-note-tip-important-caution-warning)

**Comment by lucaspar** [comment]
It seems this is the block that needs to change - if any devs want to take on this task:

https://github.com/microsoft/vscode-markdown-languageservice/blob/eb65157b610fca5e04ab44103ebea25a707add47/src/languageFeatures/documentLinks.ts#L319-L324

**Comment by mjbvz** [comment]
I'm good with adding a special case for `[!` refs. Spec wise you can write:

```md
[!NOTE]

[!NOTE]: http://example.com
```

but it's likely not common

**Comment by mjbvz** [comment]
Fixed by #191
